Doc/library/stdtypes.rst

```diff
@@ -2392,6 +2392,196 @@ that ``'\0'`` is the end of the string.
+   '}'
+   >>> F'{{ }}'
+   '{ }'
+
+A single opening curly bracket, ``'{'``, marks a replacement field that starts with
+a python expression.::
+
+   >>> f"4 * 3 is {4 * 3}"
+   '4 * 3 is 12'
+
+The above is quite contrived and a more useful use in code could be::
+
+   >>> a = 4
+   >>> b = 3
+   >>> f"The product of {a} and {b} is {a * b}"
+   'The product of 4 and 3 is 12'
+
+Or more simply::
+
+   >>> a = 4
+   >>> b = 3
+   >>> f"The product is {a * b}"
+   'The product is 12'
+
+Methods can also be used::
+
+   >>> string = "hello, world!"
+   >>> f"{string.title()}"
+   'Hello, World!'
+   >>> def make_title(input):
+   ...     return input.title()
+   ...
+   >>> f"{make_title(string)}"
+   'Hello, World!'
+
+Any variable that can be presented as a string can be used with f-strings::
+
+   >>> import datetime
+   >>> now = datetime.datetime.now()
+   >>> f"{now}"
+   '2020-07-19 23:39:01.199675'
+
+Note that this is the :func:`str` format of a :mod:`datetime`. If you want to show the
+:func:`repr` format you use a conversion. There are three conversions beginning with the
+``'!'`` (exclamation) mark:
+
++------------+---------------+
+| Conversion | Meaning       |
++============+===============+
+| ``'!r'``   | :func:`repr`  |
++------------+---------------+
+| ``'!s'``   | :func:`str`   |
++------------+---------------+
+| ``'!a'``   | :func:`ascii` |
++------------+---------------+
+
+In our example of the datetime that becomes::
+
+   >>> f"{now!r}"
+   'datetime.datetime(2020, 7, 19, 23, 39, 1, 199675)'
+   >>> f"{now!a}"
+   'datetime.datetime(2020, 7, 19, 23, 39, 1, 199675)'
+
+While debugging it may be helpful to see both the expression and its value.
+This can be shown by appending the ``'='`` (equal) after the Python expression.
+
+::
+
+   >>> f"{now=}"
+   'now=datetime.datetime(2020, 7, 19, 23, 39, 1, 199675)'
+   >>> f"{ now = }"  # Note the addition of spaces
+   ' now = datetime.datetime(2020, 7, 19, 23, 39, 1, 199675)'
+
+Note that when using the debugging operator, any additional spaces will be left intact in the output.
+Also note that by default ``'='`` displays the :func:`repr` of the expression. It is possible to show
+the :func:`str` by appending a converter::
+
+   >>> f"{now=!s}"
+   'now=2020-07-19 23:39:01.199675'
+   >>> f"{ now = !a}" # Note the additional spaces
+   ' now = datetime.datetime(2020, 7, 19, 23, 39, 1, 199675)'
+
+Once the output has been evaluated it can then be formatted using the :func:`format` protocol.
+Formatting instructions are appended preceeded by a ``':'`` (colon).
+
+::
+
+   >>> f"{now:%B %d, %Y}"
+   'July 19, 2020'
+
+The format specifier is passed to the :meth:`__format__` method of the expression or conversion result.
+An empty string is passed when the format specifier is omitted. The formatted result is then included
+in the final value of the whole string.
+
+::
+
+   >>> width = 10
+   >>> precision = 4
+   >>> import decimal
+   >>> value = decimal.Decimal("12.34567")
```

Comment:
Why not just using a float?  Some people might not be familiar with the decimal module, and wonder if this is something specifically related to decimal.
You could also add a comment next to width and precision to explain briefly what they do.

Reply:
The problem is that the difference between `repr()` and `str()` for floats are not clear. This is clearer for decimal.Decimal.

```
>>> value=12.3456789
>>> dec_value=decimal.Decimal(value)
>>> repr(value)
'12.3456789'
>>> str(value)
'12.3456789'
>>> repr(value).__format__(".3")
'12.'
>>> str(value).__format__(".3")
'12.'
>>> repr(dec_value)
"Decimal('12.345678899999999345027390518225729465484619140625')"
>>> str(dec_value)
'12.345678899999999345027390518225729465484619140625'
>>> repr(dec_value).__format__(".3")
'Dec'
>>> str(dec_value).__format__(".3")
'12.'
```

Reply:
This is better when **`value`** in `decimal.Decimal(value)` is a string of numbers.